RealmTasks Apple/RealmTasks macOS/ItemCellView.swift

```diff
@@ -0,0 +1,499 @@
+
+        get {
+            return textView.editable
+        }
+    }
+
+    var isUserInteractionEnabled = true {
+        didSet {
+            highlightView.hidden = true
+            updateTrackingAreas()
+        }
+    }
+
+    var backgroundColor: NSColor {
+        set {
+            contentView.backgroundColor = newValue
+        }
+
+        get {
+            return contentView.backgroundColor
+        }
+    }
+
+    let contentView = ColorView()
+
+    let textView: NSTextField = ItemTextField()
+    let textViewConstraintGroup = ConstraintGroup()
+
+    private let highlightView = HighlightView()
+    private let overlayView = ColorView()
+
+    private let doneIconView: NSImageView = {
+        let imageView = NSImageView()
+        imageView.image = NSImage(named: "DoneIcon")
+        return imageView
+    }()
+
+    private let deleteIconView: NSImageView = {
+        let imageView = NSImageView()
+        imageView.image = NSImage(named: "DeleteIcon")
+        return imageView
+    }()
+
+    private var releaseAction: ReleaseAction?
+
+    required init(identifier: String) {
+        super.init(frame: .zero)
+        self.identifier = identifier
+
+        setupUI()
+        setupGestures()
+
+        setTrackingAreaWithRect(bounds, options: [.MouseEnteredAndExited, .ActiveInKeyWindow])
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure(item: CellPresentable) {
+        text = item.text
+        completed = item.completed
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        alphaValue = 1
+        contentView.frame.origin.x = 0
+        editable = false
+        highlightView.hidden = true
+        isUserInteractionEnabled = true
+    }
+
+    override func updateTrackingAreas() {
+        if isUserInteractionEnabled {
+            setTrackingAreaWithRect(bounds, options: [.MouseEnteredAndExited, .ActiveInKeyWindow])
+        } else {
+            resetTrackingAreas()
+        }
+    }
+
+    private func setupUI() {
+        setupIconViews()
+        setupContentView()
+        setupBorders()
+        setupHighlightView()
+        setupOverlayView()
+        setupTextView()
+    }
+
+    private func setupIconViews() {
+        doneIconView.frame.size.width = iconWidth
+        doneIconView.frame.origin.x = iconOffset
+        doneIconView.autoresizingMask = [.ViewMaxXMargin, .ViewHeightSizable]
+        addSubview(doneIconView, positioned: .Below, relativeTo: contentView)
+
+        deleteIconView.frame.size.width = iconWidth
+        deleteIconView.frame.origin.x = bounds.width - deleteIconView.bounds.width - iconOffset
+        deleteIconView.autoresizingMask = [.ViewMinXMargin, .ViewHeightSizable]
+        addSubview(deleteIconView, positioned: .Below, relativeTo: contentView)
+    }
+
+    private func setupContentView() {
+        addSubview(contentView)
+
+        contentView.frame = bounds
```

Comment:
It's just more handy to change `contentVIew` frame directly than using `LayoutGroups` while handling swipe because there are also 2 views with `done` and `remove` icons.